coordinator/src/services/translation/response/identifiers.ts

```diff
@@ -0,0 +1,56 @@
+import {fhir} from "@models"
+
+const SPURIOUS_CODE_MATCHER = /^G[67][0-9]{6}$/
+const NMC_CODE_MATCHER = /^[0-9]{2}[A-Z][0-9]{4}[A-Z]$/
+const GMP_CODE_MATCHER = /^G[01234589][0-9]{6}$/
+const GMC_CODE_MATCHER = /^C[0-9]{7}$/
+const GPHC_CODE_MATCHER = /^[0-9]{7}$/
+const HCPC_CODE_MATCHER = /^[A-Z]{2}[0-9]{6}$/
+const DIN_CODE_MATCHER = /^[0-9]{6}$/
+
+export function looksLikeSpuriousCode(userId: string): boolean {
+  return SPURIOUS_CODE_MATCHER.test(userId)
+}
+export function looksLikeNmcCode(userId: string): boolean {
+  return NMC_CODE_MATCHER.test(userId)
+}
+export function looksLikeGmpCode(userId: string): boolean {
+  return GMP_CODE_MATCHER.test(userId)
+}
+export function looksLikeGmcCode(userId: string): boolean {
+  return GMC_CODE_MATCHER.test(userId)
+}
+export function looksLikeGphcCode(userId: string): boolean {
+  return GPHC_CODE_MATCHER.test(userId)
+}
+export function looksLikeHcpcCode(userId: string): boolean {
+  return HCPC_CODE_MATCHER.test(userId)
+}
+export function looksLikeDinCode(userId: string): boolean {
+  return DIN_CODE_MATCHER.test(userId)
+}
+
+export function createPractitionerOrRoleIdentifier(userId: string): fhir.Identifier {
```

Comment:
What you have is pretty explicit and is probably the better way to do it, but if we don't individually need the `looksLike...` functions, you could map the id and its associated schema together, which would avoid the chain of if statements:

```
identifiers = [
	{
		regex: /^G[67][0-9]{6}$/,
		schema: "https://fhir.hl7.org.uk/Id/nhsbsa-spurious-code"
	}
]

createPractitionerOrRoleIdentifier = (userId: string) => {
	for (const id in identifiers) {
		if (id.regex.test(userId)) {
			return fhir.createIdentifier(id.schema, userId)
		}
	}

	return fhir.createIdentifier("https://fhir.hl7.org.uk/Id/professional-code", userId)
}
```

Reply:
Yeah I initially thought we'd need at least some of the individual methods, but that turned out not to be the case. Happy to change if you want

Reply:
Done